common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala

```diff
@@ -22,7 +22,7 @@ import java.util.Locale
  * All structured logging `keys` used in `MDC` must be extends `LogKey`
  */
 trait LogKey {
-  val name: String = this.toString.toLowerCase(Locale.ROOT)
+  def name: String = this.getClass.getSimpleName.stripSuffix("$").toLowerCase(Locale.ROOT)
```

Comment:
shall we have a val or a lazy val to memorize the key?

Reply:
Okay, let's use `lazy val`, I have tested it and it works well.

Reply:
<img width="789" alt="image" src="https://github.com/apache/spark/assets/15246973/e9d5ee39-c1cd-431a-ac9a-648a684ddf9f">